File: QTD2.Infrastructure/Model/SimulatorScenario/SimulatorScenario_UpdateTasks_VM.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QTD2.Infrastructure.Model.SimulatorScenario
{
    public class SimulatorScenario_UpdateTasks_VM
    {
        public bool IncludeEnablingObjectives { get; set; }
        public bool IncludeProcedures { get; set; }
        public List<SimulatorScenario_Task_VM> Tasks { get; set; }

        public SimulatorScenario_UpdateTasks_VM(bool includeEnablingObjectives, bool includeProcedures)
        {
            IncludeEnablingObjectives = includeEnablingObjectives;
            IncludeProcedures = includeProcedures;
        }

        public SimulatorScenario_UpdateTasks_VM() { }
    }
}

File: QTD.SPA/src/app/_DtoModels/SimulatorScenarios_New/SimulatorScenario_UpdateTasks_VM.ts
import { SimulatorScenario_Task_VM } from "./SimulatorScenario_Task_VM";

export class SimulatorScenario_UpdateTasks_VM {
    includeEnablingObjectives: boolean;
    includeProcedures: boolean;
    tasks: SimulatorScenario_Task_VM[] = []

    setTasks(task: SimulatorScenario_Task_VM){
        var selectedTask = this.tasks.filter(r => r.taskId == task.taskId);
        if(selectedTask.length === 0){
            this.tasks.push(task);
        }
    }
}